### PR TITLE
Bring mmun/ember-bind API and implementation

### PR DESCRIPTION
**Pull request by Serabe**
This PR merges ember-bind-helper and mmun/ember-bind.

Things that need to happen for this PR to be merged:

- [x] Update README

### CODE REVIEW

**Comment by mmun** [comment]
cc @rwjblue

**Comment by rwjblue** [comment]
This looks great, and is very exciting!

File: tests/integration/components/bind-test.js [hunk 1]
```diff
@@ -2,67 +2,141 @@ import { moduleForComponent, test } from 'ember-qunit';
...
+  this.assertContext(assert, get(this, 'model.actions'));
+  this.assertArguments(assert, []);
+});
+
+test('it changes the context of a function, function: model.actions.method, context: implicit model.actions.', function(assert) {
+  this.render(hbs`<button {{action (bind model.actions.method)}}>Click</button>`);
+
+  this.callAction();
+
+  this.assertContext(assert, get(this, 'model.actions'));
```
Comment (rwjblue) [inline tests/integration/components/bind-test.js]: I think this should actually be `model`. Essentially, I believe we should special case `something.actions` to use a default target of `something`.

**Comment by Serabe** [comment]
@rwjblue I added it as last commit. Most of the changes are tests moving from `model.actions` to `model.submodel`.

I like this idea, so I'm 👍

File: README.md [hunk 2]
```diff
@@ -28,7 +28,38 @@ controller
 To solve this, you can use the `bind` helper:
 
 ```hbs
-<button onclick={{action (bind myObject.myMethod myObject)}}>
+<button onclick={{action (bind myObject.myMethod)}}>
```
Comment (mmun) [inline README.md]: There's no need to wrap this in `{{action}}`

File: lib/bind-transform.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  ```
+
+  becomes
+
+  ```hbs
+  {{bind submit x y z target=this}}
+  {{bind this.submit x y z target=this}}
+  {{bind some.service.submit x y z target=some.service}}
+  {{bind submit x y z target=existing.target}}
+  {{bind (action 'submit') x y z target=null}}
```
Comment (mmun) [inline lib/bind-transform.js]: Should probably add an example here for future code readers:

`{{bind some.service.actions.submit}}` -> `{{bind some.service.actions.submit target=some.service}}`

**Comment by mmun** [comment]
Looks great. Just a couple small tweaks in the documentation.

**Pull request merged**